src/services/wallets/wallet.ts

```diff
@@ -0,0 +1,71 @@
+import { HexString } from "../../utilities/types";
+import metamaskWallet from "./metamask";
+import torusWallet from "./torus";
+import Web3 from "web3";
+
+export enum WalletType {
+  TORUS = "TORUS",
+  METAMASK = "METAMASK",
+  // Add new WalletTypes Here
+}
+
+export interface Wallet {
+  login: () => Promise<HexString>;
+  logout: () => void;
+  getAddress: () => Promise<HexString>;
+  getBalance: (walletAddress: HexString) => Promise<number>;
```

Comment:
Do we use getBalance?

Reply:
No I left it in thinking we might end up using it. I can remove it though